From b94526da955e5412c5c86e2e4d8cc7da69868e43 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Tue, 4 Jul 2017 13:58:20 -0400
Subject: [PATCH] Memory exhaustion in PCX coder

This is a backport of upsream fix. Policy for anonymous memory does not exist.
Thus fix only corrupted pcx problem

bug-debian: https://bugs.debian.org/870491
bug: https://github.com/ImageMagick/ImageMagick/issues/536
origin: https://github.com/ImageMagick/ImageMagick/commit/3ded916c5da6febe9660c3cfa44c3114567adf74
---
 coders/pcx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/coders/pcx.c b/coders/pcx.c
index 13fad35f6..66abff873 100644
--- a/coders/pcx.c
+++ b/coders/pcx.c
@@ -359,6 +359,8 @@ static Image *ReadPCXImage(const ImageInfo *image_info,ExceptionInfo *exception)
       ThrowPCXException(CorruptImageError,"ImproperImageHeader");
     pcx_info.reserved=(unsigned char) ReadBlobByte(image);
     pcx_info.planes=(unsigned char) ReadBlobByte(image);
+    if (pcx_info.planes > 6)
+      ThrowPCXException(CorruptImageError,"ImproperImageHeader");
     if ((pcx_info.bits_per_pixel*pcx_info.planes) >= 64)
       ThrowPCXException(CorruptImageError,"ImproperImageHeader");
     if (pcx_info.planes == 0)
